### ISSUE TITLE
Regression on Windows. Folders can't be named "aux"

### ISSUE THREAD

**Issue by melund** [snakemake/snakemake#411]
@johanneskoester This recent commit 82c06e0dfd458 broke snakemake on Windows.

The name "aux" is not allowed for files and folder on Windows. Sigh.... 
```
  File "C:\Users\AzureAgent\miniconda3\envs\snakemake\lib\os.py", line 223, in makedirs
    mkdir(name, mode)
NotADirectoryError: [WinError 267] The directory name is invalid: 'C:\\actions-runner\\_work\\ANSURcompiler\\ANSURcompiler\\compiler\\.snakemake\\aux'
```
Any chance of getting regression testing activated for windows?

**Comment by johanneskoester** [comment]
What a crazy restriction. Yes, sorry for waiting so long with this. I am still a bit worried about the various places where windows needs special handling. Given that they are so actively promoting WSL, what is your opinion on that. Is there still a use case for Snakemake on plain windows at all?

**Comment by melund** [comment]
I could maybe use snakemake in WSL on my local machine and then call the relevant windows executables. But getting such a configuration a CI system running Windows is not yet possible I think.

So the answer is yes. I think there is still a usecase for snakemake on Windows. At least for the basic functionality of snakemake.

**Comment by melund** [comment]
The restricted file/folder names is a fun consequence of Windows' extreme backwards comparability:

https://www.youtube.com/watch?v=bC6tngl0PTI

**Comment by johanneskoester** [comment]
So @melund you mean that the need for windows just exists for testing Snakemake in CI, or testing your pipelines in CI? I guess the latter?

**Comment by HappyPeng2x** [comment]
I use snakemake to develop a lightweight application, internal to my company, with a GUI to launch local calculations.

So it is true that I don't use its full power, however I think it is rather convenient, and portability as well as small size were the main reasons to choose it.

For now it would be considered rather overkill by my company to allow me to set-up WSL just for this application and would make the GUI part difficult, however I think it may change in the future if integration gets better with WSL2 and the new features they are rolling out.

So @johanneskoester I understand your point but I think it is still a bit early, and I would be sad if the decision was made on this particular bug.

In this case I don't think Windows needs special handling, because it looks possible to just change the folder name so that it works on all platforms.

If issues on Windows are limited to this kind of small bugs I think it can be fixed, and though I never contributed to snakemake I would like to try to help.

Then if new features are difficult to support on Windows it can be considered at this time to offer them or not on this platform; as @melund said I think basic support is still better than no support.

**Comment by johanneskoester** [comment]
Thanks guys, that is sufficient for me to keep the support. But indeed we should make automatic windows tests work ASAP then. I will first quickly fix the aux issue in this commit and then revisit your PR @melund.

**Comment by johanneskoester** [comment]
Fixed in commit f945eef4

**Comment by gungorbudak** [comment]
Thanks much for this fix @johanneskoester when can we expect to see this fix in Conda snakemake-minimal package?
